File: 002/002.cpp
#include <iostream>

// By considering the terms in the Fibonacci sequence whose values
// do not exceed four million, find the sum of the even-valued terms.

int main() {
    int i = 1;
    int j, total = 0;
    for (int k = 0; k < 4000000; k = i+j) {
        j = i;
        i = k;
        if (k % 2 == 0) {
            total += k;
        }
    }
    std::cout << total << std::endl;
}

File: 001-010/002/002.rb
#!/usr/local/bin/ruby -w

# By considering the terms in the Fibonacci sequence whose values
# do not exceed four million, find the sum of the even-valued terms.

def main
    i = 1
    k = 0
    total = 0

    while k < 4_000_000
        j = i
        i = k
        total += k if (k % 2).zero?
        k = i + j
    end

    puts total
end

main
